### PR TITLE
Add /list_scenarios_for_plan endpoint

### PR DESCRIPTION
**Pull request by elsiel23**

### CODE REVIEW

File: src/planscape/plan/views.py [hunk 1]
```diff
@@ -520,6 +521,27 @@ def get_scenario(request: HttpRequest) -> HttpResponse:
         return JsonResponse(ScenarioSerializer(scenario_exists).data, safe=False)
     except Exception as e:
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
+    
+@csrf_exempt
+def list_scenarios_for_plan(request: HttpRequest) -> HttpResponse:
+    try:
+        assert isinstance(request.GET['plan_id'], str)
+        plan_id = request.GET.get('plan_id', "0")
+        plan_exists = Plan.objects.get(id=plan_id)
```
Comment (strangeties) [inline src/planscape/plan/views.py:530]: Maybe this was already addressed in a previous pr:
1) Can this variable be named plan?
2) and is it possible to call get_user followed by get_plan_by_id?

File: src/planscape/plan/tests.py [hunk 2]
```diff
@@ -1228,3 +1228,56 @@ def test_get_scenario_ok(self):
...
+    def test_list_does_not_belong_to_user(self):
+        not_owned_plan = Plan.objects.create(owner=None)
+        self.client.force_login(self.user)
+        response = self.client.get(
+            reverse('plan:list_scenarios_for_plan'),
+            {'plan_id': not_owned_plan.pk},
+            content_type="application/json")
+        self.assertEqual(response.status_code, 400)
+
+    def test_list_scenario_ok(self):
```
Comment (strangeties) [inline src/planscape/plan/tests.py:1276]: nit: should something explicit be returned if a plan exists and the user can see it, but the plan has zero scenarios?

File: src/planscape/plan/views.py (hunk 1 above)
Comment (elsiel23) [reply src/planscape/plan/views.py]: 1. renamed
2. this method uses a different name for the id field, so leaving things as is for now.

Comment (strangeties) [reply src/planscape/plan/views.py]: makes sense!
I have a preference for consistent names for url parameters (maybe get_plan_by_id should use "plan_id", too) for all entry points.
but that's a TODO for another day.

**Review by strangeties — APPROVED** [review]

**Comment by elsiel23** [comment]
Mixed up my merge orders. The primary list_scenario changes were merged as part of #556. Merging followups.

**Pull request merged**